[metadata] add `host_aliases` setting

vickenty opened the pull request:
### What does this PR do?

This setting can be used to add custom host aliases in addition to aliases obtained from cloud providers automatically.

### Describe your test plan

Run the agent with `host_aliases` enabled in the config file (or `DD_HOST_ALIASES` env var) and verify that the configured aliases appear on the host panel in the app.

vickenty committed:
[metadata] add `host_aliases` setting

This setting can be used to add custom host aliases in addition to
aliases obtained from cloud providers automatically.

hush-hush commented on pkg/metadata/host/host.go:
```diff
@@ -106,7 +106,7 @@ func GetPythonVersion() string {
 // getHostAliases returns the hostname aliases from different provider
 // This should include GCE, Azure, Cloud foundry, kubernetes
 func getHostAliases() []string {
-	aliases := []string{}
+	aliases := config.Datadog.GetStringSlice("host_aliases")
```
We should add a test around this new behavior. Also do we dedup aliases somewhere ? I don't think we needed before as we would not get multiple time the same alias but we might now.

hush-hush replied:
Also, I think host aliases must respect RFC1123 like hostname does (to double check). If it's a case we need to apply the check here too a drop the invalid ones,

vickenty replied:
> We should add a test around this new behavior.

I considered adding a test for `getHostAliases()` as there isn't any now, but the kublet provider talks to external system and isn't configurable. This means the test result would potentially depend on the environment, which I think isn't a good behavior for a unit test. Perhaps there is an obvious way to refactor this to make testing easer that I am missing?

> Also do we dedup aliases somewhere ? I don't think we needed before as we would not get multiple time the same alias but we might now.

Good point, added.

> Also, I think host aliases must respect RFC1123 like hostname does (to double check). If it's a case we need to apply the check here too a drop the invalid ones,

Looks like the back end is happy with whatever we send it (arbitrary ascii, unicode, emojis), so I'm not sure there is anything we could reject.

apigirl reviewed (approved):
content looks fine

vickenty committed:
[metadata] deduplicate collected host aliases

Addition of `host_aliases` introduces a chance that duplicate host
aliases will be sent to the back end. Remove duplicates to reduce
network load.

vickenty committed:
[metadata] validate host aliases

Make sure that host aliases confirm to RFC1123 before we send them to
the backend.

ogaca-dd commented on pkg/metadata/host/host.go:
```diff
@@ -150,7 +151,16 @@ func getHostAliases() []string {
-	return aliases
+	validatedAliases := []string{}
+	for _, alias := range util.SortUniqInPlace(aliases) {
+		if err := validate.ValidHostname(alias); err == nil {
```
Did you check all host aliases are valid hostnames for all providers?

vickenty replied:
All external providers appear to return aliases that are valid hostnames, but you have a good point. It would be safer to apply validation only to the aliases configured by the user.

vickenty committed:
[metadata] don't validate external host aliases

Avoid potential compatibility issues by only applying validation to
the configured custom host aliases.
